cli/azd/pkg/tools/terraform/terraform.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package terraform
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/executil"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/blang/semver/v4"
+)
+
+type TerraformCli interface {
+	tools.ExternalTool
+	Validate(ctx context.Context, file string) (string, error)
+	RunCommand(ctx context.Context, args ...string) (executil.RunResult, error)
+}
```

Comment:
Similar to our other CLI wrappers - I would like to see the primary commands called out on the interface with their most popular params called out.  This will make it much easier for future developers that are not terraform experts to call into the terraform CLI in a repeatable manner.

Reply:
addressed